src/robot/variables/assigner.py

```diff
@@ -158,21 +203,29 @@ def resolve(self, return_value):
 
 class OneReturnValueResolver:
 
-    def __init__(self, variable):
-        self._variable = variable
+    def __init__(self, assignment):
+        match: VariableMatch = search_variable(assignment, ignore_errors=True)
+        # _assignment contains the tuple: (variable, tuple of corresponding selectors)
+        self._assignment = (match.name, match.items)
 
     def resolve(self, return_value):
         if return_value is None:
-            identifier = self._variable[0]
+            identifier = self._assignment[0][0]
             return_value = {'$': None, '@': [], '&': {}}[identifier]
-        return [(self._variable, return_value)]
+        # Return the list of tuples ([variables, item selectors, rv])
+        return [(*self._assignment, return_value)]
 
 
 class _MultiReturnValueResolver:
 
-    def __init__(self, variables):
-        self._variables = variables
-        self._min_count = len(variables)
+    def __init__(self, assignments):
+        # _assignment contains a list of
+        # ... tuples: (variable, tuple of corresponding selectors)
+        self._assignments = []
+        for assigment in assignments:
+            match: VariableMatch = search_variable(assigment, ignore_errors=True)
+            self._assignments.append((match.name, match.items))
```

Comment:
I got a feeling saving names and items to separate lists would be better. At least the default `_resolve` below could use just `return list(zip(self._names, self._items, return_value))`, but I'm not sure what that would do for `ScalarsAndListReturnValueResolver`.

Reply:
I've done as you said for both `_MultiReturnValueResolver` and `ScalarsAndListReturnValueResolver`. See next comment.